asmcomp/arm64/emit.mlp

```diff
@@ -1166,7 +1168,10 @@ let begin_assembly() =
   let lbl_begin = Compilenv.make_symbol (Some "code_begin") in
   emit_named_text_section lbl_begin;
   `	.globl	{emit_symbol lbl_begin}\n`;
-  `{emit_symbol lbl_begin}:\n`
+  `{emit_symbol lbl_begin}:\n`;
+  (* avoid collision for unwind test with code_begin symbol *)
+  if macosx then ` nop\n .align 3\n`;
```

Comment:
There may be other reasons to add a `nop` here, see https://github.com/ocaml/ocaml/issues/4690.  But why `.align 3`?

Reply:
Added the reference to #4690.
The align is needed because the linker would otherwise giving warnings about `shared_startup__code_{begin,end}` for `tests/lib-dynlink-pr4839`. These warnings cause the test to fail.
I've updated the comment for these two points in 996e206